### PR TITLE
Port django-mobile to Django 2.2/3.

### PR DESCRIPTION
**Pull request by Kaniabi**
Porting django-mobile to Django 2.2/3.

This is a one way port. No longer compatible with "Django<2".

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,32 +1,19 @@
...
-deploy:
-  provider: pypi
-  user: gremu
-  password:
-    secure: mT6Gzp14P5GuWUi0MlXiBZuH6pb6M6daPe350BLo+66DQ/s1RMqqBU3lu7KTaNIKui1ZoitfNyiMiQ0QYP3+4RvR2/9HK9BEqVY7lK5mDxRWxDr1lVUnV3Bg37Vqj8792xdstDbRwP7/EkZEuviTTWAMEbYE8YJu4M/XLWpgnYY=
-  on:
-    tags: true
-    repo: gregmuellegger/django-mobile
-    condition: "$TOX_ENV = py34-18"
+#deploy:
```
Comment (utek) [inline .travis.yml:11]: We can remove that

File: django_mobile/__init__.py [hunk 2]
```diff
@@ -53,7 +54,8 @@ def get_backend(self):
         if not settings.FLAVOURS_STORAGE_BACKEND:
             raise ImproperlyConfigured(
                 u"You must specify a FLAVOURS_STORAGE_BACKEND setting to "
-                u"save the flavour for a user.")
+                u"save the flavour for a user."
```
Comment (utek) [inline django_mobile/__init__.py:57]: we are porting to django 2/3 which are py3 only. We can remove all the u"

File: .travis.yml [hunk 3]
```diff
@@ -1,32 +1,14 @@
...
+      env: TOX_ENV=py36
+    - python: 3.7
+      env: TOX_ENV=py37
+    - python: 3.8
+      env: TOX_ENV=py38
 before_install:
   - sudo pip install tox
 script:
   - tox -e $TOX_ENV
-deploy:
```
Comment (samupl) [inline .travis.yml:24]: Aren't we supposed to have our own deployment script? (I'm fine if we're going to do the deployment on a separate PR, just please make this clear)

File: django_mobile/loader.py [hunk 4]
```diff
@@ -14,48 +14,39 @@ class Loader(BaseLoader):
...
-                try:
-                    return loader.load_template_source(
-                        template_name,
-                        template_dirs)
-                except TemplateDoesNotExist:
-                    pass
-        raise TemplateDoesNotExist("Tried %s" % template_name)
+    def _prepare_template_name(self, template_name):
+        """Prepare the template name taking into consideration the flavor."""
+        template_name = "%s/%s" % (get_flavour(), template_name)
```
Comment (samupl) [inline django_mobile/loader.py:46]: Maybe let's not use old string formatting, while we're at it

File: django_mobile_tests/test_base.py [hunk 5]
```diff
@@ -7,29 +7,32 @@
...
+def str_p3_response(string):
     """
     Since response.content is a binary string in python 3,
     we decode it to make it comparable to str objects
     ( python 2 compatibility )
     """
-    if IS_PYTHON_3 :
-        return string.decode( 'ASCII' )
+    if IS_PYTHON_3:
+        return string.decode("ASCII")
```
Comment (samupl) [inline django_mobile_tests/test_base.py:32]: If we're supporting only Django 2.2+, then I think we can skip checking for python3 alltogether, as 2 is no longer supported.

File: pytest.ini [hunk 6]
```diff
@@ -0,0 +1,5 @@
+[pytest]
+norecursedirs = .git
+
+DJANGO_SETTINGS_MODULE=django_mobile_tests.settings
+DJANGO_CONFIGURATION=
```
Comment (samupl) [inline pytest.ini:5]: Do we need this var, if it's empty?

File: .travis.yml (hunk 3 above)
Comment (Kaniabi) [reply .travis.yml:24]: I'm doing the deploy manually, if that's ok?

File: pytest.ini (hunk 6 above)
Comment (Kaniabi) [reply pytest.ini:5]: We need the environment variable empty but this value is ignored by pytest, so I'm removing.
Defining DJANGO_CONFIGURATION makes pytest-django think that we are using django-configurations, but we are not. At least not here.

File: django_mobile/loader.py (hunk 4 above)
Comment (Kaniabi) [reply django_mobile/loader.py:46]: I'm trying to avoid making "improvements" in this code to avoid introducing new bugs. I'm not sure how well this code is covered by tests.
